### ISSUE TITLE
[bug] NCCL Error configurations in tasks, anyway to expose shm options in a task?

### ISSUE THREAD

**Issue by Davidnet** [kubeflow/pipelines#9147]
### Environment

*  How do you deploy Kubeflow Pipelines (KFP)?
Vertex AI Pipelines

*  KFP SDK version:
$ kfp --version
kfp 2.0.0-beta.13

### Steps to reproduce

Hi kubeflow pipelines team, I am using Vertex AI Pipelines to train detection models, and a since today my pipelines are failing due to a NCCL error on GPU machines that have more than one GPU, I am creating the task as follows:

```python
@dsl.container_component
def train_model(
    container_trainer_uri: str,
    train_dataset: Input[Dataset],
    test_dataset: Input[Dataset],
    categories_json: Input[Artifact],
    pretrained_model: Input[Artifact],
    num_iters: int,
    batch_size: int,
    lr: str,
    num_epochs: int,
    lr_step: str,
    gpus: str,
    num_workers: int,
    val_intervals: int,
    model_output: Output[Artifact],
    model_metrics: Output[Markdown],
):
    import os

    python_cmd = (
        "python3 main.py ctdet "
        " --dataset vertex_db "
        f"--train_path {str(train_dataset.path)} "
        f"--val_path {str(test_dataset.path)} "
        f"--categories_json {categories_json.path} "
        f"--load_model {pretrained_model.path} "
        "--exp_id vertex_training "
        "--input_res 512 "
        f"--num_iters {num_iters} "
        f"--batch_size {batch_size} "
        f"--lr {lr} "
        f"--num_epochs {num_epochs} "
        f"--lr_step {lr_step} "
        f"--gpus {gpus} "
        f"--num_workers {num_workers} "
        f"--val_intervals {val_intervals} "
        f"--markdown_path {model_metrics.path} "
        f"--output_path {model_output.path} "
    )

    return dsl.ContainerSpec(
        image=str(container_trainer_uri),
        command=["sh", "-c", python_cmd],
    )


train_task = train_model(
        container_trainer_uri=container_uri.output,
        train_dataset=train_importer_task.output,
        test_dataset=test_importer_task.output,
        categories_json=categories_importer_task.output,
        pretrained_model=pretrained_model.output,
        num_iters=num_iters,
        batch_size=batch_size,
        lr=lr,
        num_epochs=num_epochs,
        lr_step=lr_step,
        gpus=gpus,
        num_workers=num_workers,
        val_intervals=val_intervals,
    )
    train_task.set_display_name("Train model").set_cpu_limit("12").set_memory_limit(
        "170G"
    ).add_node_selector_constraint("NVIDIA_TESLA_A100").set_gpu_limit(
        "2"
    ).set_env_variable(
        name="NCCL_SHM_DISABLE", value="1" # Set or unset does not work
    )
```

Then since April 12th, all my pipelines start to fail with:

```
NCCL Error 2: unhandled system error
```

This I solved usually by augmenting the shared memory or allowing `ipc=host` on the host, the thing is that with kubeflow pipeline v2. `ContainerSpec` does not have control over this, neither that like mounting a volume to shm. The whole thing also is that probably something in the infrastructure on Google changed, and it broke my training steps. So any help in this?

### Expected result
Container runs without issues.

### Labels

/area components

---

Impacted by this bug? Give it a 👍.

**Comment by connor-mccarthy** [comment]
@hilcj, could you PTAL?

/assign @hilcj

**Comment by arturnn** [comment]
I am experiencing the same problem in Vertex AI Pipelines. Have you been able to create a workaround for this issue?

**Comment by Davidnet** [comment]
@hilcj ?

**Comment by chensun** [comment]
This appears to be a Vertex Pipelines backend issue. @Davidnet do you have access to GCP custom support? If so, please do file a ticket through that channel.
Also it looks like you have a workaround via Google Cloud Pipelines Components. 

I'm closing this issue as there's not much we can do from Kubeflow Pipelines open source side.
